dissect/target/filesystems/vbk.py

```diff
@@ -0,0 +1,105 @@
+from __future__ import annotations
+
+import logging
+import stat
+from pathlib import Path
+from typing import BinaryIO, Iterator
+
+from dissect.archive import vbk
+
+from dissect.target.exceptions import (
+    NotASymlinkError,
+)
+from dissect.target.filesystem import (
+    Filesystem,
+    FilesystemEntry,
+    VirtualFilesystem, )
+from dissect.target.helpers import fsutil
+
+log = logging.getLogger(__name__)
+
+
+class VBKFilesystem(Filesystem):
+    """Filesystem implementation for VBK files.
+    """
+
+    __type__ = "vbk"
+
+    def __init__(
+            self,
+            fh: BinaryIO,
+            *args,
+            **kwargs,
+    ):
+        super().__init__(fh, *args, **kwargs)
+
+        self.vbk = vbk.VBK(fh)
+        self._fs = VirtualFilesystem(alt_separator=self.alt_separator, case_sensitive=self.case_sensitive)
+
+        def map_entries(directory, parent_path) -> None:
+            for entry in directory.iterdir():
+                entry_path = parent_path.joinpath(entry.name)
+                if entry.is_dir():
+                    map_entries(entry, entry_path)
+                else:
+                    self._fs.map_file_entry(entry_path.__str__(), VBKFilesystemEntry(self, str(entry_path), entry))
+
+        map_entries(self.vbk.get("/"), Path("/"))
```

Comment:
The VBK implementation already has it's own filesystem-like methods and path resolution, no need to walk the entire VBK and map everything into a virtual filesystem, just call the relevant methods on the VBK implementation.

You can look at other "real" filesystem implementations in `dissect.target` for inspiration.